lib/git-util.js

```diff
@@ -18,6 +18,67 @@ exports.latestTagOrFirstCommit = async () => {
 };
 
 exports.hasUpstream = async () => {
-	const {stdout} = await execa('git', ['status', '--short', '--branch', '--porcelain=2']);
+	const stdout = await execa.stdout('git', ['status', '--short', '--branch', '--porcelain=2']);
 	return /^# branch\.upstream [\w\-/]+$/m.test(stdout);
 };
+
+exports.currentBranch = () => execa.stdout('git', ['symbolic-ref', '--short', 'HEAD']);
+
+exports.isWorkingTreeClean = async () => {
+	try {
+		const status = await execa.stdout('git', ['status', '--porcelain']);
+		if (status !== '') {
+			return false;
+		}
+
+		return true;
+	} catch (_) {
+		return false;
+	}
+};
+
+exports.remoteHistoryClean = async () => {
+	let stdout;
+	try { // Gracefully handle no remote set up.
+		stdout = await execa.stdout('git', ['rev-list', '--count', '--left-only', '@{u}...HEAD']);
+	} catch (_) {}
+
+	if (stdout && stdout !== '0') {
+		return false;
+	}
+
+	return true;
+};
+
+exports.validRemote = async () => {
+	try {
+		await execa('git', ['ls-remote', 'origin', 'HEAD']);
+		return true;
+	} catch (error) {
+		return error.stderr.replace('fatal:', 'Git fatal error:');
+	}
+};
+
+exports.fetchRemote = () => execa('git', ['fetch']);
```

Comment:
This is a pretty small command which could just stay in `lib/prerequisite.js`, but since we talked about moving _all_ Git commands into a separate util file, I moved this as well.

Reply:
👍 Makes sense to me.